src/main/java/net/sf/jabref/model/metadata/MetaData.java

```diff
@@ -173,139 +92,117 @@ public AbstractBibtexKeyPattern getBibtexKeyPattern(GlobalBibtexKeyPattern globa
      *
      * @param bibtexKeyPattern the key patterns to update to. <br />
-     *                     A reference to this object is stored internally and is returned at getBibtexKeyPattern();
+     *                     A reference to this object is stored internally and is returned at getCiteKeyPattern();
      */
-    public void setBibtexKeyPattern(AbstractBibtexKeyPattern bibtexKeyPattern) {
-        // remove all keypatterns from metadata
-        Iterator<String> iterator = this.iterator();
-        while (iterator.hasNext()) {
-            String key = iterator.next();
-            if (key.startsWith(PREFIX_KEYPATTERN)) {
-                iterator.remove();
-            }
-        }
+    public void setCiteKeyPattern(AbstractBibtexKeyPattern bibtexKeyPattern) {
+        Objects.requireNonNull(bibtexKeyPattern);
+
+        List<String> defaultValue = bibtexKeyPattern.getDefaultValue();
+        Map<String, List<String>> nonDefaultPatterns = bibtexKeyPattern.getPatterns();
+        setCiteKeyPattern(defaultValue, nonDefaultPatterns);
+    }
+
+    public void setCiteKeyPattern(List<String> defaultValue, Map<String, List<String>> nonDefaultPatterns) {
+        // Remove all patterns from metadata
+        citeKeyPatterns.clear();
 
-        // set new value if it is not a default value
-        Set<String> allKeys = bibtexKeyPattern.getAllKeys();
-        for (String key : allKeys) {
-            if (!bibtexKeyPattern.isDefaultValue(key)) {
-                List<String> data = new ArrayList<>();
-                data.add(bibtexKeyPattern.getValue(key).get(0));
-                String metaDataKey = PREFIX_KEYPATTERN + key;
-                this.putData(metaDataKey, data);
-            }
+        // Set new value if it is not a default value
+        for (Map.Entry<String, List<String>> pattern : nonDefaultPatterns.entrySet()) {
+            citeKeyPatterns.put(pattern.getKey(), pattern.getValue().get(0));
         }
 
-        // store default pattern
-        if (bibtexKeyPattern.getDefaultValue() == null) {
-            this.remove(KEYPATTERNDEFAULT);
+        // Store default pattern
+        if (defaultValue.isEmpty()) {
+            defaultCiteKeyPattern = null;
         } else {
-            List<String> data = new ArrayList<>();
-            data.add(bibtexKeyPattern.getDefaultValue().get(0));
-            this.putData(KEYPATTERNDEFAULT, data);
+            defaultCiteKeyPattern = defaultValue.get(0);
         }
 
-        this.bibtexKeyPattern = bibtexKeyPattern;
+        postChange();
     }
 
-    public Optional<List<String>> getSaveActions() {
-        return Optional.ofNullable(getData(SAVE_ACTIONS));
+    public Optional<FieldFormatterCleanups> getSaveActions() {
+        return Optional.ofNullable(saveActions);
     }
 
     public Optional<BibDatabaseMode> getMode() {
-        List<String> data = getData(DATABASE_TYPE);
-        if ((data == null) || data.isEmpty()) {
-            return Optional.empty();
```

Comment:
If `mode` is null, then still an empty optional is returned (Optional.ofNullable(mode))